plugin/pkg/shoot/validator/admission.go

```diff
@@ -244,21 +260,30 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, _ adm
 	}
 
 	var secretBinding *gardencorev1beta1.SecretBinding
-	if a.GetOperation() == admission.Create && shoot.Spec.SecretBindingName != nil {
```

Comment:
Why was this condition removed?

Reply:
The check for `Create` verb was removed because one can now change between `SecretBindingName` and `CredentialsBindingName`, i.e. the field is not immutable.